### ISSUE TITLE
rootdirectory

### ISSUE THREAD

**Issue by castillo-n** [microweber/microweber#278]
I'm wondering if you could implement something that would help me (us) to have microweber in the root directory and just have the css, js and images in the public directory?
Thank you! I appreciate your amazing work!

**Comment by ash-rain** [comment]
You could always go with Laravel's default folder structure (a dedicated "public" directory) but you'll have to delete this line or change it to whatever you like
https://github.com/microweber/microweber/blob/1.0/src/Microweber/MicroweberServiceProvider.php#L148

**Comment by ash-rain** [comment]
Perhaps it would make sense to make that easier to configure but that's not a priority for now. The reason for flattening the folder structure are most shared hosting providers which don't let users change their public directories. So yeah, not very awesome but compatibility has always been in the way of perfection :)

**Comment by castillo-n** [comment]
Man! Thank you! I would try it after I get back, I appreciate it! Own you a beer, glass of wine, coffee or tea, or bottle of water at least :)

**Comment by ash-rain** [comment]
Yes, we drink all of these and several others :) Have a great day

**Comment by ash-rain** [comment]
Just keep in mind we're not done with the final release yet so any changes to the core could cause you to make manual merges when updating.

**Comment by castillo-n** [comment]
ok, thanks!

**Comment by ash-rain** [comment]
Actually I recommend you copying that line to 
https://github.com/microweber/microweber/blob/1.0/src/Microweber/MicroweberServiceProvider.php#L148
and adding 'public' or whatever after base_path(). It would be safer.

**Comment by castillo-n** [comment]
it didnt work :( my public folder is public_html but I tried to concat it like this: base_path()."public_html", base_path()."/public_html",  return base_path()."public_html",  return base_path()."/public_html",  but i didnt work.

**Comment by ash-rain** [comment]
Did you also move `index.php` and the other public files from the base directory? Where did you make the changes? In the `AppServiceProvider` or `MicroweberServiceProvider`?

**Comment by castillo-n** [comment]
I add it to AppServiceProvider and as you said in the first message I just commented out the whole line in MicroweberServiceProvider/ , which are the files that I have to move to the public folder? I thought that with this I was going to be able to have everything except for css, js, and picture folders in the public folder

**Comment by ash-rain** [comment]
Nope, this will change your public folder. That means you'll also have to have `index.php` and `.htaccess` in that folder. Any "browsable" files at all. Now I realized you'll also have to change some paths in that same `index.php`.

You don't have to delete the line from `MicroweberServiceProvider` if you're calling the method in `AppServiceProvider`, you do either one of these but not necessarily both.

However, I haven't tested it out so I'm not 1000% sure if it's going to be that simple. Let me know how it goes. We'll surely do something about making it easier in the future.

**Comment by ash-rain** [comment]
Actually, what's the matter with the current setup? Your assets (CSS, JS) should be in `userfiles` in your template's folder.

**Comment by castillo-n** [comment]
Thanks for helping me. what I did is just what I was toughed  to do,(dont leave any unnecessary file in the public folder, and put everything in the root folder.) so what I did is, I made a index php in the public and I routed it to the one you made by default. I created a css, js, and img folder in the public, and since then I have tried to make it work like that. The userfiles folder is in my root directory (When I say root directory I mean the parent of my public_html folder). And there isnt anything wrong, it is just that I cant use microweber until I put it out of the reach of people, in the same way as laravel is by default

**Comment by ash-rain** [comment]
I understand. However, this shouldn't be an issue if you've secured your web server properly. I mean, if someone hacks into your server and is able to read files having a designated public folder won't make a difference.
With that said, we'll further look into this.

**Comment by castillo-n** [comment]
I appreciate it

**Comment by ash-rain** [comment]
We'll put this in the config for the next release. You'll be able to change the public folder from there.

**Comment by Staalkoper** [comment]
I believe @peter-mw has experience with this - the base of the solution is to use symbolic links pointing to the root folder (usr/share/microweber).

**Comment by peter-mw** [comment]
hi 

i still haven't got to this issue, for now all the sensitive folders are protected by `.htaccess` file in each of them 

i agree that changing public folder has to be done, so will i leave this issue open until its resolved.
